src/tests/ftest/util/server_utils.py

```diff
@@ -634,11 +635,16 @@ def check_rank_state(self, ranks, valid_state, max_checks=1):
                 # The regex failed to get the rank and state
                 raise ServerFailed("Error obtaining {} output: {}".format(self.dmg, data))
             checks += 1
+            failed_ranks = []
             for rank in ranks:
-                if data[rank]["state"] == valid_state:
-                    return [True]
-
-        return [False, rank]
+                if data[rank]["state"] not in valid_states:
+                    failed_ranks.append(rank)
+                else:
+                    counter += 1
+            if counter == len(ranks):
+                return []
```

Comment:
What's the purpose of `counter`? Can't we just do this?
```suggestion
            if not failed_ranks:
                return []
```
Also, the method description `Check the state of single rank in DAOS system.` is confusing to me because it looks like we're checking multiple ranks, not single rank.

Reply:
Purpose of counter is to limit the number of checks. Say for example, if the method call asked for 5 max checks, but the first round of check itself returned the desired output, the method won't waste time running the remaining 4 checks.

Reply:
I think Makito means we don't need the `counter` variable for this. `counter == len(ranks)` is only `True` if `failed_ranks == []`

Reply:
Ah okay. Yeah I get what you guys mean now. I have made the modifications and re-pushed. Thanks for pointing it out.